plugins/vpp/l2plugin/vppcalls/dump_vppcalls.go

```diff
@@ -125,6 +137,52 @@ func (handler *BridgeDomainVppHandler) DumpBridgeDomainIDs() ([]uint32, error) {
 	return activeDomains, nil
 }
 
+// Reads ARP termination table from all bridge domains. Result is then added to bridge domains.
+func (h *BridgeDomainVppHandler) dumpBridgeDomainMacTable() (map[uint32][]*l2nb.BridgeDomains_BridgeDomain_ArpTerminationEntry, error) {
+	defer func(t time.Time) {
+		h.stopwatch.TimeLog(l2ba.BridgeDomainDump{}).LogTimeEntry(time.Since(t))
+	}(time.Now())
+
+	bdArpTable := make(map[uint32][]*l2nb.BridgeDomains_BridgeDomain_ArpTerminationEntry)
+	req := &l2ba.BdIPMacDump{BdID: ^uint32(0)}
+
+	reqCtx := h.callsChannel.SendMultiRequest(req)
+	for {
+		msg := &l2ba.BdIPMacDetails{}
+		stop, err := reqCtx.ReceiveReply(msg)
+		if err != nil {
+			return nil, err
+		}
+		if stop {
+			break
+		}
+
+		// Prepare ARP entry
+		arpEntry := &l2nb.BridgeDomains_BridgeDomain_ArpTerminationEntry{}
+		var ipAddr net.IP
+		if uintToBool(msg.IsIPv6) {
+			ipAddr = msg.IPAddress
```

Comment:
This could directly call `.To16()` since that returns same type. Then you could move next line outside if for both cases and call `arpEntry.IpAddress = ipAddr.String()`.